Support explicit syntax

delaneyj opened the pull request:
Death to the $

khalwat committed:
refactor: Use Docker-dev for the base image, and build all the things

khalwat committed:
refactor: Log in to the registry

khalwat committed:
refactor: Run on workflow changes

khalwat committed:
chore: revert changes

bencroker committed:
Improve homepage title

delaneyj committed:
delete everything first

delaneyj committed:
add top level sdk folder
move go SDK
add back bundles and stuff to make tsbuild work

delaneyj committed:
update go SDK url

khalwat committed:
fix: Fix issue with `make task`

khalwat committed:
fix: Fix multiple linux arch’s, add `support` as a `deploy` dep

khalwat committed:
refactor: Install `flyctl`

khalwat committed:
refactor: Output to the current directory

khalwat committed:
refactor: ignore .out

khalwat committed:
refactor: Run via Docker container

khalwat committed:
refactor: Use `flyctl` instead of the alias

khalwat committed:
refactor: Docker-in-Docker

As per “The Socket Solution” here: https://jpetazzo.github.io/2015/09/03/do-not-use-docker-in-docker-for-ci/

khalwat committed:
refactor: Fix comment

khalwat committed:
refactor: Make fly alias

khalwat committed:
refactor: Use fly alias

delaneyj committed:
add vendored back in

khalwat committed:
fix: Revert relative paths

khalwat committed:
refactor: Revert ignoring `.out`

khalwat committed:
refactor: Run flyctl in the CI container

khalwat committed:
refactor: Remove flyctl

khalwat committed:
refactor: Remove socket bind mount

khalwat committed:
refactor: Add `flyctl` back in so `make task deploy` can be done locally

delaneyj committed:
add types and nested signals

bencroker committed:
Add canonical meta tag

delaneyj committed:
add more stuff back in

delaneyj committed:
add ( i think ) the rest of the plugins

bencroker commented on library/src/plugins/official/backend/actions/sse.ts:
```diff
@@ -0,0 +1,121 @@
+    fn: async (
+        { el: { id: elId }, signals },
+        url: string,
+        { method = "GET", headers = {}, onlyRemote = true }: {
```
`method` should be converted to uppercase somewhere.

bencroker commented on library/src/bundles/datastar.ts:
Reorder plugins.

bencroker commented:
Reminder to add the ability to filter `data-on-signals-changed` based on a signal path in dot notation.

Also, change the argument that `setAll()` and `toggleAll()` take from a regexp to a path in dot notation.

delaneyj committed:
lazy generation ReactiveExpresison, get click to edit working

bencroker commented:
Reminder to switch to the following modifier syntax, to enable nestable signals in attribute keys.

```
data-signals-foo.bar?mod1_arg1_arg2&mod2_arg1_arg2=""
```

delaneyj committed:
data-merge-signals to data-signals
start @action to action

bencroker committed:
Add WIP release notes

delaneyj committed:
update more examples removing @ syntax and moving to sse action

bencroker committed:
Merge branch 'delaneyj/issue296' of https://github.com/starfederation/datastar into delaneyj/issue296

delaneyj committed:
remove deprecated examples

delaneyj committed:
Merge branch 'delaneyj/issue296' of github.com:starfederation/datastar into delaneyj/issue296

delaneyj committed:
The great $ purge of 2024

bencroker committed:
Add error codes

delaneyj committed:
working on v21 syntax  #296
fix bind/intersect lazy reactiveExpression gen

bencroker committed:
Merge branch 'develop' into delaneyj/issue296

bencroker committed:
Fix todo demo

bencroker committed:
Reorder plugins

delaneyj committed:
Merge branch 'delaneyj/issue296' of github.com:starfederation/datastar into delaneyj/issue296

delaneyj committed:
Merge branch 'delaneyj/issue296' of github.com:starfederation/datastar into delaneyj/issue296

bencroker committed:
Start markdown errors

bencroker committed:
Finish markdown errors

bencroker committed:
Update release notes

bencroker committed:
Add release note

bencroker committed:
Update release note

delaneyj committed:
remove dup and require checks

delaneyj committed:
fix signals JSON usage
fix throttle/debounce lead/trail port

delaneyj committed:
make js obj helper
auto formatting
signal value needed explicit period prefix for value
data-signals lazy generates due to possible expressons
removeSignal applies after eval

delaneyj committed:
update test, 127/127

bencroker committed:
Merge branch 'delaneyj/issue296' of https://github.com/starfederation/datastar into delaneyj/issue296

bencroker committed:
Create VSCode extension (#295)

* Add attributes

* Finish attributes

* Fix repo URL

* Finish

delaneyj committed:
change attributes

delaneyj committed:
Merge branch 'delaneyj/issue296' of github.com:starfederation/datastar into delaneyj/issue296

delaneyj committed:
fix attributes

bencroker committed:
VSCode extension improvements

bencroker committed:
Merge branch 'delaneyj/issue296' of https://github.com/starfederation/datastar into delaneyj/issue296

delaneyj committed:
fix data-attributes example

bencroker committed:
Fix a BEN!

delaneyj committed:
support individual data-class-foo calls

delaneyj committed:
Merge branch 'delaneyj/issue296' of github.com:starfederation/datastar into delaneyj/issue296

delaneyj committed:
error page

bencroker committed:
Rework error pages

bencroker committed:
Simplify

bencroker committed:
Rework errors

bencroker committed:
Fixes

bencroker committed:
Delete `console.log` from library

delaneyj committed:
#296 persist back in

delaneyj committed:
Merge branch 'delaneyj/issue296' of github.com:starfederation/datastar into delaneyj/issue296

bencroker committed:
Add missing error

delaneyj committed:
remove mouse move until I implement correctly

delaneyj committed:
i hate javascript

delaneyj committed:
. syntax for modifiers now :

bencroker committed:
Extract plugin rules

bencroker committed:
Add clarification comment

delaneyj committed:
support xor key or value for bind/ref/indicator

bencroker committed:
Merge branch 'delaneyj/issue296' into extract-plugin-rules

# Conflicts:
#	bundles/datastar.js
#	bundles/datastar.js.map
#	code/dotnet/sdk/src/Consts.fs
#	code/php/sdk/src/Consts.php
#	library/src/plugins/official/backend/attributes/indicator.ts
#	library/src/plugins/official/dom/attributes/bind.ts
#	library/src/plugins/official/dom/attributes/ref.ts
#	sdk/go/consts.go

bencroker committed:
Add KeyValRules

bencroker committed:
Remove unused properties

bencroker committed:
Update VSCode extension

bencroker committed:
Update docs

bencroker committed:
Improve getting started guide

delaneyj committed:
logic restored

delaneyj committed:
Merge branch 'delaneyj/issue296' of github.com:starfederation/datastar into delaneyj/issue296

delaneyj committed:
autogen error sidebar links

delaneyj committed:
fixed Restructure folders #287

delaneyj committed:
fix scroll test

bencroker committed:
Prep docs for release

bencroker committed:
Fixed check for exclusive requirements

bencroker committed:
Build

delaneyj committed:
img diff

delaneyj committed:
add errors to nested signals
fix getting started
